### ISSUE TITLE
Training stalls at torch.tensor(ids, device=device)

### ISSUE THREAD

**Issue by shijian2001** [wdrink/SimpleAR#25]
Hi team,  

I'm encountering an issue where the training code gets stuck indefinitely at the following line: `completion_ids = [torch.tensor(ids, device=device) for ids in completion_ids]`. There are no error messages, but the process remains unresponsive until a DDP timeout occurs.  

**Context:**  
- I've made two modifications to the code:  
  1. Added a custom reward function (though this likely isn't the cause, as the stall occurs before reward calculation)  
  2. Implemented a dynamic data sampling strategy (this also seems unrelated, as model generation work fine in previous steps)  

**Observed Behavior:**
The issue occurs when running on 8x H800 GPUs with the following GPU status during the hang:
```python
[0] NVIDIA L20Y | 32°C, 100 % | 54491 / 81559 MB |
[1] NVIDIA L20Y | 41°C, 100 % |  7571 / 81559 MB |
[2] NVIDIA L20Y | 42°C, 100 % |  7571 / 81559 MB |
[3] NVIDIA L20Y | 34°C, 100 % |  7571 / 81559 MB |
[4] NVIDIA L20Y | 34°C, 100 % |  7571 / 81559 MB |
[5] NVIDIA L20Y | 42°C, 100 % |  7571 / 81559 MB |
[6] NVIDIA L20Y | 40°C, 100 % |  7411 / 81559 MB |
[7] NVIDIA L20Y | 30°C,   0 % | 59499 / 81559 MB |
```

**Questions:**  
1. How can I better debug this hanging issue?  
2. Are there any known scenarios that might cause this behavior?  

Thank you for your help!

**Comment by wdrink** [comment]
Hi @shijian2001 , is your self-implemented reward a rule-based one or a model-based one? If it is a model-based one, I will recommend you to debug the forward pass of the reward function (for example, commenting it and see if the program still hangs)

**Comment by shijian2001** [comment]
I've been using model-based rewards in my training setup, and I previously resolved the program hanging issue by adding the environment variables `NCCL_P2P_DISABLE=1` and `NCCL_IB_DISABLE=1`.

However, I'm now facing a new problem when training on 8 GPUs (but it works well when single-GPU training, and one GPU for VLLM). The error occurs at the line:
```python
completion_ids = broadcast_object_list(completion_ids, from_process=0)
```
The error message is `pickle data was truncated.` This happens before even entering the reward calculation steps.

Have you encountered similar issues? @wdrink

**Comment by wdrink** [comment]
haven't ran into this issue before, could you please share the full error message, thanks!

**Comment by wdrink** [comment]
Closed as inactivity, feel free to re-open it if needed.
